### PR TITLE
Pal 641 create notices file

### PR DESCRIPTION
**Pull request by pd104923**

### CODE REVIEW

**Review by ac74475 — COMMENTED** [review]
Looks really good, Makes it much easier to see when we have changed versions of dependancies or even have multiple versions. E.g caffeine and eureka.

**Review by ac74475 — CHANGES_REQUESTED** [review]
Is it possible to remove the generation of the html licence files?

**Comment by ac74475** [comment]
doesn't appear to be easily possible within the plugin as it downloads all the licences as per the link that the dependency stated its licence link to be. Therefore some projects have defined that as a .txt and some as .html files.

**Review by ac74475 — DISMISSED** [review]

**Review by dev930018 — DISMISSED** [review]

**Comment by dev930018** [comment]
@ac74475 There has been a preference to link the notices.md to the txt license if available

File: data-service/pom.xml [hunk 1]
```diff
@@ -26,11 +26,13 @@
...
     </parent>
 
+    <!--  *********** Aritfact ID ***********    -->
     <artifactId>data-service</artifactId>
+    <url>https://github.com/gchq/Palisade-services/tree/develop/data-service</url>
+    <!--  *********** Aritfact ID ***********    -->
 
     <properties>
         <eureka.version>1.9.16</eureka.version>
-        <databind.version>2.9.8</databind.version>
```
Comment (dev930018) [inline data-service/pom.xml]: Why has this gone?

File: policy-service/pom.xml [hunk 2]
```diff
@@ -38,7 +41,7 @@
         <dependency>
             <groupId>com.github.ben-manes.caffeine</groupId>
             <artifactId>caffeine</artifactId>
-            <version>${caffeine.version}</version>
+            <version>2.8.1</version>
```
Comment (dev930018) [inline policy-service/pom.xml]: Should be reverted back to caffeine.version

File: user-service/pom.xml [hunk 3]
```diff
@@ -25,11 +25,15 @@
...
 
+    <!--  *********** Aritfact ID ***********    -->
     <artifactId>user-service</artifactId>
+    <url>https://github.com/gchq/Palisade-services/tree/develop/user-service</url>
+    <!--  *********** Aritfact ID ***********    -->
 
     <properties>
         <spring-cloud-starter-ribbon.version>1.4.7.RELEASE</spring-cloud-starter-ribbon.version>
         <swagger.version>1.5.21</swagger.version>
+        <eureka.version>1.9.16</eureka.version>
```
Comment (dev930018) [inline user-service/pom.xml]: No point adding this as it's getting removed in PAL-653

File: user-service/pom.xml [hunk 4]
```diff
@@ -75,6 +79,7 @@
         <dependency>
             <groupId>com.netflix.eureka</groupId>
             <artifactId>eureka-client</artifactId>
+            <version>${eureka.version}</version>
```
Comment (dev930018) [inline user-service/pom.xml]: ditto, revert

**Review by dev930018 — CHANGES_REQUESTED** [review]

**Review by developer01189998819991197253 — COMMENTED** [review]
Approved after previous comments are addressed

File: data-service/pom.xml (hunk 1 above)
Comment (pd104923) [reply data-service/pom.xml]: There was a duplication of the Jackson Databind license information. This value was the older version and was not being used within the pom so I removed it.

File: user-service/pom.xml (hunk 3 above)
Comment (pd104923) [reply user-service/pom.xml]: Will remove this.

File: policy-service/pom.xml (hunk 2 above)
Comment (pd104923) [reply policy-service/pom.xml]: This was using version 2.8.0 from the `spring-boot-starter-parent 2.2.0.RELEASE`. Added 2.8.1 as a property in the pom to override this.

Comment (pd104923) [reply policy-service/pom.xml]: This has also been done in the user-service pom as well.

File: user-service/pom.xml (hunk 4 above)
Comment (pd104923) [reply user-service/pom.xml]: Reverted

**Review by m78233 — APPROVED** [review]

**Review by dev930018 — DISMISSED** [review]

**Review by nw1984 — DISMISSED** [review]

**Review by m78233 — DISMISSED** [review]

**Review by ac74475 — DISMISSED** [review]

**Review by developer01189998819991197253 — DISMISSED** [review]

**Review by ac74475 — APPROVED** [review]

**Pull request merged**